File: src/main/kotlin/io/paulocosta/twitbooks/goodreads/Results.java
package io.paulocosta.twitbooks.goodreads;

import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;

@Root(name = "results", strict = false)
public class Results {

    public Results() {}

    @ElementList(name = "work", inline = true)
    private List<Work> works;

    public List<Work> getWorks() {
        return works;
    }

    public void setWorks(List<Work> works) {
        this.works = works;
    }
}

File: backend/src/main/kotlin/io/paulocosta/twitbooks/books/provider/goodreads/Results.kt
package io.paulocosta.twitbooks.books.provider.goodreads;

import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

@Root(name = "results", strict = false)
data class Results @JvmOverloads constructor(
        @field:ElementList(name = "work", inline = true, required = false)
        @param:ElementList(name = "work", inline = true, required = false)
        var works: List<Work> = emptyList()
)
